addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -266,7 +266,7 @@ export class Wysiwyg extends Component {
                     isDifferentRecord
                 )
             {
-                if (isDifferentRecord) {
+                if (isDifferentRecord || !newProps.options.record.dirty) {
```

Comment:
I'm not sure I understand the new fix, though. We suddenly receive the _same_ record as before but somehow it is dirty now while it wasn't before? This looks like a weird case.

Reply:
The idea behind the new fix is that when clicking discard, we set dirty as `false` here  https://github.com/odoo/odoo/blob/a9d2771d3ae19c0685d904ab908c6400e7f2630a/addons/web/static/src/model/relational_model/record.js#L534 
So if it's a different record or the field is not dirty we reset the editor otherwise we update the value only. I don't know if there is a flow where the field is not dirty and we need to reset too

Reply:
I see, thanks for the explanation. However, it's impossible to understand that just from reading the code... ^^' What do you think of this?
```js
const isDiscardedRecord = !isDifferentRecord && !newProps.options.record.dirty;
if (isDifferentRecord || isDiscardedRecord) {
```
If you don't like it because we test `isDifferentRecord` twice then I guess a comment would work too.

Reply:
@dmo-odoo , I updated the fix here, can you take a look ?  thanks !